### PR TITLE
feat(pie-link): DSW-1056 create basic functionality of pie link

### PR DESCRIPTION
**Pull request by raoufswe**
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 1]
```diff
@@ -1,23 +1,81 @@
...
+        size: {
+            description: 'Set the size of the link.',
+            control: 'select',
+            options: sizes,
+            defaultValue: {
+                summary: 'medium',
+            },
+        },
+        href: {
+            description: 'What the URL that the hyperlink should point to',
```
Comment (FayeCarter) [inline apps/pie-storybook/stories/pie-link.stories.ts:47]: I think just 'The URL that the hyperlink should point to'

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 2]
```diff
@@ -30,13 +88,40 @@ const linkStoryMeta: LinkStoryMeta = {
...
-const Template = ({ }: LinkProps): TemplateResult => html`
-  <pie-link></pie-link>
-`;
+const Template = ({
+    href, target, rel, size, variant, isBold, isStandalone, slot,
+}: LinkProps): TemplateResult => html`
+        <pie-link
+            variant="${variant}"
+            size="${size}"
+            href="${ifDefined(href || undefined)}"
```
Comment (FayeCarter) [inline apps/pie-storybook/stories/pie-link.stories.ts:97]: Can these be `... || nothing`?

File: packages/components/pie-link/README.md [hunk 3]
```diff
@@ -57,9 +57,15 @@ import { PieLink } from '@justeattakeaway/pie-link/dist/react';
...
 
-| Property | Type | Default | Description |
-|---|---|---|---|
-| - | - | - | - |
+| Property     | Type        | Default       | Description                                                                 |
+| ------------ | ----------- | ------------- | --------------------------------------------------------------------------- |
+| variant      | `String`  | `default`   | Variant of the link, one of variants – default, high-visibility or inverse |
+| size         | `String`  | `medium`    | Size of the link, one of `sizes` – `medium`, `small`                 |
+| href         | `String`  | `undefined` | Native html `href` attribute                                              |
+| rel          | `String`  | `undefined` | native html `rel` attribute                                               |
```
Comment (FayeCarter) [inline packages/components/pie-link/README.md:65]: nit, sorry. But can all of these start with a capital letter please? Sorry

File: apps/pie-storybook/stories/pie-link.stories.ts (hunk 2 above)
Comment (jamieomaguire) [reply apps/pie-storybook/stories/pie-link.stories.ts:97]: I'd look into the `?` and `.` attribute modifiers in Lit templating

File: packages/components/pie-link/README.md (hunk 3 above)
Comment (jamieomaguire) [reply packages/components/pie-link/README.md:65]: @FayeCarter do you mean the descriptions? I agree if so but I think we should keep props camel case

File: apps/pie-storybook/stories/pie-link.stories.ts (hunk 2 above)
Comment (raoufswe) [reply apps/pie-storybook/stories/pie-link.stories.ts:97]: `... ||` nothing works like a charm, thank youuu

File: packages/components/pie-link/README.md (hunk 3 above)
Comment (raoufswe) [reply packages/components/pie-link/README.md]: fixed

**Review by jamieomaguire — APPROVED** [review]

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 4]
```diff
@@ -1,23 +1,80 @@
...
 const keptReferences = [PieLink];
 
+type LinkProps = SlottedComponentProps<LinkBaseProps>;
 type LinkStoryMeta = StoryMeta<LinkProps>;
 
 const defaultArgs: LinkProps = {
-
+    variant: 'default',
+    size: 'medium',
+    href: 'https://pie.design',
```
Comment (FayeCarter) [inline apps/pie-storybook/stories/pie-link.stories.ts:18]: 🔥

**Pull request merged**